### DISCUSSION TITLE
Put options from pytest.ini in conftest.py instead

### DISCUSSION THREAD

**Discussion by sylvorg** [pytest-dev/pytest#11264] [Q&A] [answered]
Hello!

I'd like to modify the what test names and files are collected, and whether their `AssertionErrors` are diffed or not. Is there any way to put these in the `conftest.py` file instead of the `pytest.ini` file?

Thank you kindly for the help!

**Comment by RonnyPfannschmidt** [comment]
No, conftests has a different role and is not suitable for the config

**Reply by sylvorg** [reply]
Ah; so is there no way to modify what tests are collected through the conftest, not just what files?

**Reply by RonnyPfannschmidt** [reply]
Conftests have documented way's to affect collection in their belonging folders that structurally differs from the global configuration, without a actual concrete use case it's hard to give a reasonable recommendation

**Reply by sylvorg** [reply]
Would you be able to help me with [this question](https://stackoverflow.com/questions/76799444/no-diffs-from-pytest-assertionerrors-when-straying-from-common-test-file-name-sc/76799824) I posted on StackOverflow?

**Reply by RonnyPfannschmidt** [reply, marked as answer]
ok, so theres multiple issues with that one that are way over my time buget

1. it seems to be using hpy and i am completely unaware of its behaviours
2. its using non-python module names - (test-) and its not clear to me how that will help

my first recommendation would be to use a pytest config file that adds extra globs for the non-python tests
my second recommendation is to coordinate with the hpy community to figure how to add the interactions

**Reply by sylvorg** [reply]
Got it. I was asking one of the maintainers of hpy and they advised to simply rename the files, which I had named to follow the hpy variable naming convention, and to add globs for their so-called `mangled` names, which allow for extra characters in variable names.

Thanks for all the help!
